src/vmm/src/resources.rs

```diff
@@ -278,3 +278,302 @@ impl VmResources {
+                is_root_device: false,
+                partuuid: Some("0eaa91a0-01".to_string()),
+                is_read_only: false,
+                rate_limiter: Some(RateLimiterConfig::default()),
+            })
+            .unwrap();
+
+        block_cfgs
+    }
+
+    fn default_boot_cfg() -> BootConfig {
+        let mut kernel_cmdline = kernel::cmdline::Cmdline::new(4096);
+        kernel_cmdline.insert_str(DEFAULT_KERNEL_CMDLINE).unwrap();
+        let tmp_file = TempFile::new().unwrap();
+        BootConfig {
+            cmdline: kernel_cmdline,
+            kernel_file: File::open(tmp_file.as_path()).unwrap(),
+            initrd_file: Some(File::open(tmp_file.as_path()).unwrap()),
+        }
+    }
+
+    fn default_vm_resources() -> VmResources {
+        VmResources {
+            vm_config: VmConfig::default(),
+            boot_config: Some(default_boot_cfg()),
+            block: default_block_cfgs(),
+            network_interface: default_net_cfgs(),
+            vsock: None,
+        }
+    }
+
+    impl PartialEq for BootConfig {
+        fn eq(&self, other: &Self) -> bool {
+            self.cmdline.as_str().eq(other.cmdline.as_str())
+                && self.kernel_file.metadata().unwrap().st_ino()
+                    == other.kernel_file.metadata().unwrap().st_ino()
+                && self
+                    .initrd_file
+                    .as_ref()
+                    .unwrap()
+                    .metadata()
+                    .unwrap()
+                    .st_ino()
+                    == other
+                        .initrd_file
+                        .as_ref()
+                        .unwrap()
+                        .metadata()
+                        .unwrap()
+                        .st_ino()
+        }
+    }
+
+    #[test]
+    fn test_vcpu_config() {
+        let vm_resources = default_vm_resources();
+        let expected_vcpu_config = VcpuConfig {
+            vcpu_count: vm_resources.vm_config().vcpu_count.unwrap(),
+            ht_enabled: vm_resources.vm_config().ht_enabled.unwrap(),
+            cpu_template: vm_resources.vm_config().cpu_template,
+        };
+
+        let vcpu_config = vm_resources.vcpu_config();
+        assert_eq!(vcpu_config, expected_vcpu_config);
+    }
+
+    #[test]
+    fn test_vm_config() {
+        let vm_resources = default_vm_resources();
+        let expected_vm_cfg = VmConfig::default();
+
+        assert_eq!(vm_resources.vm_config().clone(), expected_vm_cfg);
+    }
+
+    #[test]
+    fn test_set_vm_config() {
+        let mut vm_resources = default_vm_resources();
+        let mut aux_vm_config = VmConfig {
+            vcpu_count: Some(32),
+            mem_size_mib: Some(512),
+            ht_enabled: Some(true),
+            cpu_template: Some(CpuFeaturesTemplate::T2),
+        };
+
+        assert_ne!(vm_resources.vm_config, aux_vm_config);
+        vm_resources.set_vm_config(aux_vm_config.clone()).unwrap();
```

Comment:
nit: maybe make `VmResources::set_vm_config` take a reference to vm_config instead of full ownership and avoid cloning.